gst/nnstreamer/tensor_sparse/tensor_sparse_dec.c

```diff
@@ -0,0 +1,410 @@
+  self = GST_TENSOR_SPARSE_DEC (parent);
+
+  GST_DEBUG_OBJECT (self, "Received %s query: %" GST_PTR_FORMAT,
+      GST_QUERY_TYPE_NAME (query), query);
+
+  switch (GST_QUERY_TYPE (query)) {
+    case GST_QUERY_CAPS:
+    {
+      GstCaps *caps;
+      GstCaps *filter;
+
+      gst_query_parse_caps (query, &filter);
+      caps = gst_tensor_sparse_dec_query_caps (self, pad, filter);
+      silent_debug_caps (self, filter, "filter");
+      gst_query_set_caps_result (query, caps);
+      gst_caps_unref (caps);
+      return TRUE;
+    }
+    case GST_QUERY_ACCEPT_CAPS:
+    {
+      GstCaps *caps;
+      GstCaps *template_caps;
+      gboolean res = FALSE;
+
+      gst_query_parse_accept_caps (query, &caps);
+      silent_debug_caps (self, caps, "caps");
+
+      if (gst_caps_is_fixed (caps)) {
+        template_caps = gst_pad_get_pad_template_caps (pad);
+
+        res = gst_caps_can_intersect (template_caps, caps);
+        gst_caps_unref (template_caps);
+      }
+
+      gst_query_set_accept_caps_result (query, res);
+      return TRUE;
+      break;
+    }
+    default:
+      break;
+  }
+
+  return gst_pad_query_default (pad, parent, query);
+}
+
+/**
+ * @brief Parse caps and set tensors config
+ */
+static gboolean
+gst_tensor_sparse_dec_parse_caps (GstTensorSparseDec * self,
+    const GstCaps * caps)
+{
+  GstStructure *structure;
+  GstTensorsConfig config;
+
+  g_return_val_if_fail (caps != NULL, FALSE);
+  g_return_val_if_fail (gst_caps_is_fixed (caps), FALSE);
+
+  structure = gst_caps_get_structure (caps, 0);
+
+  if (!gst_tensors_config_from_structure (&config, structure) ||
+      !gst_tensors_config_validate (&config)) {
+    /** not fully configured */
+    GST_ERROR_OBJECT (self, "Failed to configure tensors config.\n");
+    return FALSE;
+  }
+
+  self->out_config = config;
+  return TRUE;
+}
+
+static gboolean
+gst_tensor_sparse_dec_sink_event (GstPad * pad, GstObject * parent,
+    GstEvent * event)
+{
+  GstTensorSparseDec *self;
+  self = GST_TENSOR_SPARSE_DEC (parent);
+  g_return_val_if_fail (event != NULL, FALSE);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_CAPS:
+    {
+      GstCaps *caps;
+
+      gst_event_parse_caps (event, &caps);
+      silent_debug_caps (self, caps, "caps");
+
+      if (gst_tensor_sparse_dec_parse_caps (self, caps)) {
+        return TRUE;
+      } else {
+        gst_event_unref (event);
+        return FALSE;
+      }
+      break;
+    }
+    default:
+      break;
+  }
+
+  return gst_pad_event_default (pad, parent, event);
+}
+
+/**
+ * @brief Internal function to transform the input buffer.
+ */
+static GstFlowReturn
+gst_tensor_sparse_dec_chain (GstPad * pad, GstObject * parent, GstBuffer * buf)
+{
+  GstFlowReturn ret;
+  GstTensorSparseDec *self = GST_TENSOR_SPARSE_DEC (parent);
+  GstTensorMetaInfo meta;
+  GstMemory *mem, *out_mem;
+  GstMapInfo map;
+  GstBuffer *outbuf;
+  GstTensorsInfo *info;
+  int i;
+
+  info = &self->out_config.info;
```

Comment:
Q: is it necessary to keep / update output config?

Reply:
Not necessary. All info should be inferred from incoming buffer.
This line just updated srcpad's config which is might be not configured during the nego